src/components/DropDown/DropDown.tsx

```diff
@@ -0,0 +1,140 @@
+import { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { useSearchParams } from 'react-router-dom';
+import './DropDown.scss';
+import ArrowDown from '../../images/icons/arrow_down.svg';
+import { getSearchWith } from '../../utils/searchHelper';
+import { useWindowSize } from '../../hooks/useWindowSize';
+
+export type SelectorType = 'sort-by' | 'items-on-page';
+type Props = {
+  type: SelectorType;
+};
+
+export const DropDown: React.FC<Props> = ({ type }) => {
+  const [value, setValue] = useState('All');
+  const [isOpen, setIsOpen] = useState(false);
+  const [highlightedIndex, setHighlightedIndex] = useState(0);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const itemsOnPage = searchParams.get('perPage') || 'All';
+  const sort = searchParams.get('sort') || 'All';
+
+  const screenSize = useWindowSize();
+
+  const selectOption = (option: string) => {
+    if (option !== value) {
+      setValue(option);
+    }
+  };
+
+  const isOptionSelected = (option: string) => {
+    return option === value;
+  };
+
+  useEffect(() => {
+    if (isOpen) {
+      setHighlightedIndex(0);
+    }
+  }, [isOpen]);
+
+  const widthStyle = {
+    width: '128px',
+  };
+
+  if (type === 'sort-by') {
+    widthStyle.width = '176px';
+  }
+
+  if (screenSize.width >= 320 && screenSize.width <= 639) {
+    widthStyle.width = '136px';
+  }
+
+  const dropDownTitle = type === 'sort-by' ? 'Sort by' : 'Items on page';
+
+  const sortByList: Record<string, string> = {
+    All: '',
+    Alphabet: 'name',
+    Newest: 'age',
+    Chepest: 'price',
+  };
+
+  const itemsOnPageList: Record<string, string> = {
+    All: '',
+    4: '4',
+    8: '8',
+    16: '16',
+  };
```

Comment:
No need to re create this variable on each render as they static - move out of component body